Wikipedia/Code/SavedArticlesFetcher.m

```diff
@@ -177,14 +177,14 @@ - (void)fetchAllImagesInArticle:(MWKArticle*)article failure:(WMFErrorHandler)fa
 
     NSArray<NSURL*>* URLs = [[article allImageURLs] allObjects];
     
-    [self.imageController cacheImagesWithURLsInBackground:URLs failure:failure success:success];
+    [self cacheImagesWithURLsInBackground:URLs failure:failure success:success];
 }
 
 - (void)fetchGalleryDataForArticle:(MWKArticle*)article failure:(WMFErrorHandler)failure success:(WMFSuccessHandler)success {
     WMF_TECH_DEBT_TODO(check whether on - disk image info matches what we are about to fetch)
     @weakify(self);
     [self fetchImageInfoForImagesInArticle:article failure:^(NSError *error) {
-        
+        failure(error);
```

Comment:
@joewalsh possibly - the big reason I fixed it is because in the case of your fix you are relying on the failure block being called so you can set the migration flag. 

If we don't add this the app might get in a state where the migration is never marked as complete.